scripts/merge-patches.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+set -e
+
+# Variables
+PROJECT_DIR=$(pwd)
+PACKAGE_JSON="$PROJECT_DIR/package.json"
+PATCHES_DIR="$PROJECT_DIR/patches"
+TEMP_DIR="$PROJECT_DIR/temp"
+
+function merge_patches() {
+  CONFLICTS_FOUND=false
+
+  # Iterate through the patch files
+  for PATCH_PATH in "$PATCHES_DIR"/*/*.patch; do
+    PATCH_DIR=$(basename "$(dirname "$PATCH_PATH")")
+    PATCH_FILE=$(basename $PATCH_PATH)    
+    PACKAGE_NAME=$(echo "$PATCH_FILE" | cut -d'+' -f1)
+    PACKAGE_VERSION=$(jq -r ".dependencies.\"$PACKAGE_NAME\"" "$PACKAGE_JSON")
+    echo "# Manage $PATCH_PATH"
+
+    # Check if the package version exists and if the patch needs to be updated
+    if [ "$PACKAGE_VERSION" == "null" ] || [[ "$PATCH_FILE" =~ "$PACKAGE_NAME+$PACKAGE_VERSION".patch ]]; then
+      echo "Package '$PACKAGE_NAME' not found in package.json or patch already up-to-date. Skipping patch update."
+      echo ""
+      continue
+    fi
+
+    # Create a subfolder for the current package inside the temporary directory
+    PACKAGE_TEMP_DIR="$TEMP_DIR/$PATCH_DIR"
+    if [ -d "$PACKAGE_TEMP_DIR" ]; then
+      echo "$PACKAGE_TEMP_DIR already exist, skipping patch update. Remove it to regenerate."
+      echo ""
+      continue
+    fi
+
+    mkdir -p "$PACKAGE_TEMP_DIR"
+    LOG_FILE="$PACKAGE_TEMP_DIR".log
+
+    # Install the package in the package subfolder
+    cd "$PACKAGE_TEMP_DIR"
+    echo '{ }' > package.json
+    yarn add "$PACKAGE_NAME@$PACKAGE_VERSION" >> $LOG_FILE 2>&1
+    yarn add patch-package <> $LOG_FILE 2>&1
+
+    # Apply the patch with --merge option
+    patch -p1 --no-backup-if-mismatch --input="$PATCH_PATH" --forward --merge 2>&1 >> $LOG_FILE || true
```

Comment:
I received this error on MacOs with version patch 2.5.8
`patch: unrecognized option `--merge'`